extensions/vertx-common/runtime/src/main/java/io/quarkus/vertx/common/runtime/VertxCommonRecorder.java

```diff
@@ -0,0 +1,268 @@
+package io.quarkus.vertx.common.runtime;
+
+import static io.vertx.core.file.impl.FileResolver.CACHE_DIR_BASE_PROP_NAME;
+
+import java.io.File;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.atomic.AtomicReference;
+import java.util.function.Supplier;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import io.netty.channel.EventLoopGroup;
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.runtime.IOThreadDetector;
+import io.quarkus.runtime.LaunchMode;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.ShutdownContext;
+import io.quarkus.runtime.annotations.Recorder;
+import io.vertx.core.Vertx;
+import io.vertx.core.VertxOptions;
+import io.vertx.core.eventbus.EventBusOptions;
+import io.vertx.core.file.FileSystemOptions;
+import io.vertx.core.http.ClientAuth;
+import io.vertx.core.impl.VertxImpl;
+import io.vertx.core.impl.VertxThread;
+import io.vertx.core.net.JksOptions;
+import io.vertx.core.net.PemKeyCertOptions;
+import io.vertx.core.net.PemTrustOptions;
+import io.vertx.core.net.PfxOptions;
+
+@Recorder
+public class VertxCommonRecorder {
+
+    static volatile Vertx vertx;
```

Comment:
Not a big fan of having this shared instance here. I let it like that for now as I think it should be a separate discussion.

Reply:
What do you mean by shared instance?

This is the "core" Vert.x instance that should be shared by everyone (except RestEasy, as done in @stuartwdouglas ' PR)

Reply:
I mean having a volatile instance that we more or less reinitialize. I don't know if we can do better or if we need it to be around for hot replacement.

But the `more or less` is important here because a lot of the reinitialization code was buggy.